### PR TITLE
Us22 chat part1

### PR DESCRIPTION
**Pull request by MorganeL-pro**
Added Chat Controller
Added Chat view
Added relation mentoring in profile index
Resolved pb with buttons (delete)

### CODE REVIEW

File: templates/profile/index.html.twig [hunk 1]
```diff
@@ -9,43 +9,110 @@
...
+                            <p class="email-profile">{{ app.user.student.mentoring.mentor.user.email }}</p>
+                        </div>
+                        <div class="card-information-profile">
+                            <img src="{{ asset('build/images/phone.png') }}" alt="phone profile img">
+                            <p class="phone-profile">{{ app.user.student.mentoring.mentor.user.phone }}</p>
+                        </div>
+                    </div>
+                </div>
+                <div class="btn-profile m-3">
+                    <a href="{{ path("chat", {id: app.user.id}) }}" class="btn btn-secondary m-3">Chat avec mon mentor !</a>
```
Comment (Thomas905) [inline templates/profile/index.html.twig]: button do not work

**Review by Thomas905 — COMMENTED** [review]
The form is not display when not message is send

**Review by Thomas905 — DISMISSED** [review]

File: src/Controller/ChatController.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+    //function to verify is User is null or not
+    public function checkUser(int $id): User
+    {
+        $user = $this->userRepository->find($id);
+        if (is_null($user)) {
+            throw $this->createNotFoundException("No user found with id $id.");
+        }
+        return $user;
+    }
```
Comment (RomainClair) [inline src/Controller/ChatController.php]: This seems to be what the paramConverter does
You probably should use the paramConverter in index instead

File: src/Controller/ChatController.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        if ($user->getMentor() === null && $user->getStudent() === null) {
+            throw $this->createNotFoundException('User is neither a student or a mentor.');
+        }
+        if ($user->getMentor() !== null) {
+            $mentoring = $user->getMentor()->getMentoring();
+        } elseif ($user->getStudent() !== null) {
+            $mentoring = $user->getStudent()->getMentoring();
+        }
+        return $mentoring;
+    }
```
Comment (RomainClair) [inline src/Controller/ChatController.php]: This should probably be in a Service or at least be a private method because it's definitely not a controller

**Review by RomainClair — CHANGES_REQUESTED** [review]

File: src/Controller/ChatController.php (hunk 3 above)
Comment (MorganeL-pro) [reply src/Controller/ChatController.php]: I changed for a private method because it's a small one.

**Review by Thomas905 — APPROVED** [review]

**Review by RomainClair — APPROVED** [review]

**Pull request merged**